xmrstak/misc/environment.hpp

```diff
@@ -12,35 +12,30 @@ class params;
 
 struct environment
 {
-
-	static environment& inst()
-	{
-		static environment env;
-		return env;
-	}
-	
-	environment& operator=(const environment& env)
+	static inline environment& inst(environment* init = nullptr)
 	{
-		this->pPrinter = env.pPrinter;
-		this->pglobalStates = env.pglobalStates;
-		this->pJconfConfig = env.pJconfConfig;
-		this->pExecutor = env.pExecutor;
-		this->pParams = env.pParams;
-		return *this;
-	}
+		static environment* env = nullptr;
 
+		if(env == nullptr)
+		{
+			if(init == nullptr)
+				env = new environment;
```

Comment:
Could it be that this is the reason for #51. That environment is handled as POD and the pointer will not be initilized with a `nullptr`?

Reply:
https://stackoverflow.com/a/15212351

Reply:
If this can result in a uninitialized type than we need also check all other singletons.